libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/TypeRep.hs

```diff
@@ -440,19 +781,120 @@ genSizedRep _ RewardR = arbitrary
+genSizedRep _ TagR = arbitrary
+genSizedRep _ DataHashR = arbitrary
+genSizedRep _ AddrR = arbitrary
+genSizedRep _ PCredR = arbitrary
+genSizedRep _ ShelleyTxCertR = arbitrary
+genSizedRep _ ConwayTxCertR = arbitrary
+genSizedRep _ MIRPotR = arbitrary
+genSizedRep _ IsValidR = frequency [(1, pure (IsValid False)), (9, pure (IsValid True))]
+genSizedRep _ IntegerR = arbitrary
+genSizedRep _ (ScriptsNeededR p) = case whichUTxO p of
+  UTxOShelleyToMary -> pure $ ScriptsNeededF p (ShelleyScriptsNeeded Set.empty)
+  UTxOAlonzoToConway -> pure $ ScriptsNeededF p (AlonzoScriptsNeeded [])
+genSizedRep _ (ScriptPurposeR p) = case whichTxCert p of
+  TxCertShelleyToBabbage -> arbitrary
+  TxCertConwayToConway -> arbitrary
+genSizedRep _ (TxBodyR p) =
+  case p of
+    Shelley _ -> pure (TxBodyF p (newTxBody p []))
+    Allegra _ -> pure (TxBodyF p (newTxBody p []))
+    Mary _ -> pure (TxBodyF p (newTxBody p []))
+    Alonzo _ -> pure (TxBodyF p (newTxBody p []))
+    Babbage _ -> pure (TxBodyF p (newTxBody p []))
+    Conway _ -> pure (TxBodyF p (newTxBody p []))
+genSizedRep _ BootstrapWitnessR = arbitrary
+genSizedRep _ SigningKeyR = genSigningKey
+genSizedRep _ (TxWitsR p) =
+  case p of
+    Shelley _ -> TxWitsF p <$> arbitrary
+    Allegra _ -> TxWitsF p <$> arbitrary
+    Mary _ -> TxWitsF p <$> arbitrary
+    Alonzo _ -> TxWitsF p <$> arbitrary
+    Babbage _ -> TxWitsF p <$> arbitrary
+    Conway _ -> TxWitsF p <$> arbitrary
+genSizedRep _ PayHashR = arbitrary
+genSizedRep _ (TxR p) =
+  case p of
+    Shelley _ -> TxF p <$> arbitrary
+    Allegra _ -> TxF p <$> arbitrary
+    Mary _ -> TxF p <$> arbitrary
+    Alonzo _ -> TxF p <$> arbitrary
+    Babbage _ -> TxF p <$> arbitrary
+    Conway _ -> TxF p <$> arbitrary
+genSizedRep _ ScriptIntegrityHashR = arbitrary
+genSizedRep _ AuxiliaryDataHashR = arbitrary
+genSizedRep _ GovernanceActionR = pure NoConfidence
+genSizedRep _ (WitVKeyR _) = arbitrary
+genSizedRep _ (TxAuxDataR p) = genTxAuxDataF p
 genSizedRep _ CommColdHashR = arbitrary
 genSizedRep _ CommHotHashR = arbitrary
+genSizedRep _ LanguageR = arbitrary
+genSizedRep _ (LedgerStateR p) = case p of
+  Shelley _ -> arbitrary
+  Allegra _ -> arbitrary
+  Mary _ -> arbitrary
+  Alonzo _ -> arbitrary
+  Babbage _ -> arbitrary
+  Conway _ -> arbitrary
+genSizedRep _ StakeHashR = arbitrary
+genSizedRep _ BoolR = arbitrary
+genSizedRep _ DRepR = arbitrary
+genSizedRep _ (PoolMetadataR p) =
+  if restrictHash p
+    then PoolMetadata <$> arbitrary <*> (BS.take (hashsize p) <$> arbitrary)
+    else PoolMetadata <$> arbitrary <*> arbitrary
 
 genRep ::
   Era era =>
   Rep era b ->
   Gen b
 genRep x = do (NonNegative n) <- arbitrary; genSizedRep n x
 
+-- | A Byron address is made of a 32 byte ByteString, so here we
+--   generate some random ones with size 32 Bytes
+gen32ByteString :: Gen ByteString
+gen32ByteString = do
+  list <- vectorOf 32 (elements ["1", "2", "3", "4", "5", "6", "7", "8", "9", "0"])
+  pure (List.foldl' (<>) mempty list)
```

Comment:
Here is a generator you can use for generating random bytestrings: https://github.com/input-output-hk/cardano-ledger/blob/a4895859484ed34498c80c1d737dc2c4501f6fee/libs/cardano-ledger-binary/testlib/Test/Cardano/Ledger/Binary/Arbitrary.hs#L229-L230
```suggestion
gen32ByteString = genByteString 32
```

Reply:
Thank you for pointing this out.